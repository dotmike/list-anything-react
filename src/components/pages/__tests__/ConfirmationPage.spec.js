import configureStore from "redux-mock-store";

import ConnectedConfirmationPage, {
  UnconnectedConfirmationPage
} from "../ConfirmationPage";

const testData = {
  token: "some token"
};

const setup = (propOverrides = {}) => {
  const props = {
    match: { params: { token: testData.token } },
    confirm: jest.fn(() => Promise.resolve()),
    ...propOverrides
  };

  return {
    props,
    connectedWrapperShallow: wrapperShallow(ConnectedConfirmationPage, props),
    wrapperShallow: wrapperShallow(UnconnectedConfirmationPage, props)
  };
};

describe("ConnectedConfirmationPage", () => {
  const mockStore = configureStore();
  const state = {};
  it("renders correctly", () => {
    const { connectedWrapperShallow: wrapper } = setup({
      store: mockStore(state)
    });
    expect(wrapper()).toMatchSnapshot();
  });
});

describe("UnconnectedConfirmationPage", () => {
  it("renders correctly before `confirm` finishes", () => {
    const { wrapperShallow: wrapper, props } = setup();
    expect(wrapper()).toMatchSnapshot();
    expect(props.confirm).toHaveBeenCalledWith(testData.token);
  });

  it("renders correctly after `confirm` finishes successfully", done => {
    const { wrapperShallow: wrapper, props } = setup();

    wrapper();
    expect(props.confirm).toHaveBeenCalledWith(testData.token);

    setImmediate(() => {
      wrapper().update();
      expect(wrapper()).toMatchSnapshot();
      done();
    });
  });

  it("renders correctly after `confirm` finishes with failure", done => {
    const { wrapperShallow: wrapper, props } = setup({
      confirm: jest.fn(() => Promise.reject())
    });

    wrapper();
    expect(props.confirm).toHaveBeenCalledWith(testData.token);

    setImmediate(() => {
      wrapper().update();
      expect(wrapper()).toMatchSnapshot();
      done();
    });
  });
});
