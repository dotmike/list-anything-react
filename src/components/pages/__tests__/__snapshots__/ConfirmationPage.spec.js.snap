// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedConfirmationPage renders correctly 1`] = `
<ConfirmationPage
  confirm={[Function]}
  match={
    Object {
      "params": Object {
        "token": "some token",
      },
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`UnconnectedConfirmationPage renders correctly after \`confirm\` finishes successfully 1`] = `
<CenterElemsContainer>
  <Message
    icon={true}
    success={true}
  >
    <Icon
      as="i"
      name="checkmark"
    />
    <MessageContent>
      <MessageHeader>
        Thank you. Your account has been verified.
      </MessageHeader>
      <Link
        replace={false}
        to="/dashboard"
      >
        Go to your dashboard
      </Link>
    </MessageContent>
  </Message>
</CenterElemsContainer>
`;

exports[`UnconnectedConfirmationPage renders correctly after \`confirm\` finishes with failure 1`] = `
<CenterElemsContainer>
  <Message
    icon={true}
    negative={true}
  >
    <Icon
      as="i"
      name="warning sign"
    />
    <MessageContent>
      <MessageHeader>
        Ooops. Invalid token it seems.
      </MessageHeader>
    </MessageContent>
  </Message>
</CenterElemsContainer>
`;

exports[`UnconnectedConfirmationPage renders correctly before \`confirm\` finishes 1`] = `
<CenterElemsContainer>
  <Message
    icon={true}
  >
    <Icon
      as="i"
      loading={true}
      name="circle notched"
    />
    <MessageHeader>
      Validating your email
    </MessageHeader>
  </Message>
</CenterElemsContainer>
`;
